taf/git.py

```diff
@@ -401,6 +419,9 @@ def branch_exists(
         a remote branch exists.
         """
         repo = self.pygit_repo
+        if repo is None:
+            # TODO default to _git
+            return False
```

Comment:
Do we want this code to raise `GitError`? It looks like we're raising errors on `if repo is None`, but not here?

Reply:
I think the only reason is that this was the first such mypy error that I tried addressing and I initially thought that pygit2 instantiaon errors would get raised somewhere, then realized that that would not be the case and that we would return incorrect values from these functions.